packages/dts-generator/lib/phases/json-fixer.js

```diff
@@ -42,12 +43,30 @@ function addImplicitNamespaces(json) {
 const badData = {
   "sap.ui.test.Opa5": true
 };
+
 function removeBadData(json) {
   json.symbols = _.reject(json.symbols, sym => {
     return badData[sym.name];
   });
 }
 
+/**
+ * In case a interface is tagged as restricted, we should remove it from the classes that uses it
+ * @param json
+ */
+function removeRestrictedInterface(json) {
+
+  const restrictedInterfaces = _(json.symbols)
+      .filter(symbol => symbol.kind === "interface" && symbol.visibility === "restricted")
```

Comment:
Not horrible, But I recommend watching this:
https://www.youtube.com/watch?v=ecekSCX3B4Q&feature=youtu.be&t=1660